geomstats/information_geometry/exponential.py

```diff
@@ -168,14 +171,224 @@ def squared_dist(self, point_a, point_b, **kwargs):
+        n_initial_points = initial_point.shape[0]
+        n_initial_tangent_vecs = initial_tangent_vec.shape[0]
+        if n_initial_points > n_initial_tangent_vecs:
+            raise ValueError(
+                "There cannot be more initial points than " "initial tangent vectors."
+            )
+        if n_initial_tangent_vecs > n_initial_points > 1:
+            raise ValueError(
+                "For several initial tangent vectors, "
+                "specify either one or the same number of "
+                "initial points."
+            )
+
+        base = gs.exp(initial_tangent_vec / initial_point)
+
+        def path(t):
+            """Generate parameterized function for geodesic curve.
+
+            Parameters
+            ----------
+            t : array-like, shape=[n_times,]
+                Times at which to compute points of the geodesics.
+
+            Returns
+            -------
+            geodesic : array-like, shape=[..., n_times, 1]
+                Values of the geodesic at times t.
+            """
+            t = gs.reshape(gs.array(t), (-1,))
+            _base, _t = gs.broadcast_arrays(base, gs.transpose(t))
+            return gs.expand_dims(initial_point * _base**_t, axis=-1)
+
+        return path
+
+    def _geodesic_bvp(self, initial_point, end_point):
+        """Solve geodesic boundary problem.
+
+        Compute the parameterized function for the geodesic starting at
+        initial_point and ending at end_point.
+
+        Parameters
+        ----------
+        initial_point : array-like, shape=[..., 1]
+            Initial point.
+        end_point : array-like, shape=[..., 1]
+            End point.
+
+        Returns
+        -------
+        path : function
+            Parameterized function for the geodesic curve starting at
+            initial_point and ending at end_point.
+        """
+        n_initial_points = initial_point.shape[0]
+        n_end_points = end_point.shape[0]
+
+        if n_initial_points > n_end_points > 1:
+            raise ValueError(
+                "For several initial points, specify either"
+                "one or the same number of end points."
+            )
+        if n_end_points > n_initial_points > 1:
+            raise ValueError(
+                "For several end points, specify either "
+                "one or the same number of initial points."
+            )
+
+        base = end_point / initial_point
+
+        def path(t):
+            """Generate parameterized function for geodesic curve.
+
+            Parameters
+            ----------
+            t : array-like, shape=[n_times,]
+                Times at which to compute points of the geodesics.
+
+            Returns
+            -------
+            geodesic : array-like, shape=[..., n_times, 1]
+                Values of the geodesic at times t.
+            """
+            t = gs.reshape(gs.array(t), (-1,))
+            _base, _t = gs.broadcast_arrays(base, gs.transpose(t))
```

Comment:
variable names. no need to transpose `t`.

Reply:
By "variables names" you mean I should change `_base` and `_t` right?